src/probnum/filtsmooth/statespace/preconditioner.py

```diff
@@ -0,0 +1,62 @@
+"""Coordinate changes in state space models."""
+
+import abc
+
+try:
+    # cached_property is only available in Python >=3.8
+    from functools import cached_property
+except ImportError:
+    from cached_property import cached_property
+
+import numpy as np
+import scipy.special  # for vectorised factorial
+
+
+class Preconditioner(abc.ABC):
+    """
+    Coordinate change transformations as preconditioners in state space models.
+
+    For some models, this makes the filtering and smoothing steps more numerically stable.
+    """
+
+    @abc.abstractmethod
+    def __call__(self, step) -> np.ndarray:
+        # if more than step is needed, add them into the signature in the future
+        raise NotImplementedError
+
+    @cached_property
+    def inverse(self) -> "Preconditioner":
+        raise NotImplementedError
+
+
+class TaylorCoordinates(Preconditioner):
```

Comment:
Do you like this name? I had `NordsieckCoordinates` at first, but since they are different and only similar in spirit, I chose this more general name. In fact, `TaylorCoordinates` may also be wrong, but general enough that (hopefully) no one will complain...

Reply:
TaylorCoordinates sounds like it is an established thing already. Would you write "Taylor coordinates" in a paper too? But I don't really know about any better naming here neither, I'm fine with whatever you come up with.

I do like the docstring though :D

Reply:
Indeed, that sounds like an established thing. How about `TaylorLikeCoordinates` or `NordsieckLikeCoordinates` to clarify what they are without lying? In view of the docstring, how about `BetterThanNordsieckCoordinates`? (just kidding... or am I?) In a paper I'd write rescaled coordinates or something along those lines.

Reply:
I like your proposal of `NordsieckLikeCoordinates`, since the way I understand it they are mostly motivated by those Nordsieck coordinates, but improved upon for our setting. I could be easily convinced for another name though